File: mobile-app/src/screens/AttendanceScreen.js
import { Alert, Text, View } from "react-native";
import { Screen } from "../components/Screen";
import { Card } from "../components/Card";
import { PrimaryButton } from "../components/PrimaryButton";
import { markAttendance } from "../api/endpoints";
import { getCurrentPosition } from "../services/location";
import { colors, spacing } from "../theme/tokens";

export function AttendanceScreen() {
  async function submitAttendance() {
    try {
      const position = await getCurrentPosition();
      await markAttendance({
        status: "present",
        latitude: position.coords.latitude,
        longitude: position.coords.longitude,
      });
      Alert.alert("Attendance sent", "Your attendance was submitted.");
    } catch (error) {
      Alert.alert("Attendance failed", error.message);
    }
  }

  return (
    <Screen>
      <Text style={{ color: colors.text, fontSize: 28, fontWeight: "900" }}>Attendance</Text>
      <Card>
        <Text style={{ color: colors.textDark, fontWeight: "900", fontSize: 18 }}>GPS attendance</Text>
        <Text style={{ color: colors.mutedDark, marginBottom: spacing.md }}>Submit your current location to the shared Django attendance API.</Text>
        <PrimaryButton title="Mark attendance" onPress={submitAttendance} />
      </Card>
    </Screen>
  );
}

File: mobile-app-flutter/lib/screens/attendance_screen.dart
import 'package:flutter/material.dart';
import 'package:geolocator/geolocator.dart';
import '../api/endpoints.dart';
import '../theme/app_theme.dart';
import '../widgets/app_card.dart';
import '../widgets/app_screen.dart';
import '../widgets/primary_button.dart';

class AttendanceScreen extends StatefulWidget {
  const AttendanceScreen({super.key});

  @override
  State<AttendanceScreen> createState() => _AttendanceScreenState();
}

class _AttendanceScreenState extends State<AttendanceScreen> {
  bool _loading = false;
  String? _lastMessage;
  bool _lastSuccess = false;

  Future<void> _submit() async {
    setState(() {
      _loading = true;
      _lastMessage = null;
    });
    try {
      final permission = await Geolocator.checkPermission();
      if (permission == LocationPermission.denied) {
        final req = await Geolocator.requestPermission();
        if (req == LocationPermission.denied ||
            req == LocationPermission.deniedForever) {
          throw Exception(
              'Location permission is required to mark attendance.');
        }
      }
      final pos = await Geolocator.getCurrentPosition(
        locationSettings:
            const LocationSettings(accuracy: LocationAccuracy.high),
      );
      await markAttendance({
        'status': 'present',
        'latitude': pos.latitude,
        'longitude': pos.longitude,
      });
      setState(() {
        _lastSuccess = true;
        _lastMessage = 'Attendance marked successfully.';
      });
    } catch (e) {
      setState(() {
        _lastSuccess = false;
        _lastMessage = e.toString();
      });
    } finally {
      if (mounted) setState(() => _loading = false);
    }
  }

  @override
  Widget build(BuildContext context) {
    return AppScreen(
      children: [
        const Text(
          'Attendance',
          style: TextStyle(
              color: AppColors.text,
              fontSize: 28,
              fontWeight: FontWeight.w900),
        ),
        AppCard(
          children: [
            const Text(
              'GPS attendance',
              style: TextStyle(
                  color: AppColors.textDark,
                  fontSize: 18,
                  fontWeight: FontWeight.w900),
            ),
            const Text(
              'Submit your current location to mark yourself present.',
              style: TextStyle(color: AppColors.mutedDark),
            ),
            if (_lastMessage != null)
              Container(
                padding: const EdgeInsets.all(12),
                decoration: BoxDecoration(
                  color: _lastSuccess
                      ? AppColors.success.withValues(alpha: 0.1)
                      : AppColors.danger.withValues(alpha: 0.1),
                  borderRadius: BorderRadius.circular(10),
                ),
                child: Text(
                  _lastMessage!,
                  style: TextStyle(
                    color: _lastSuccess ? AppColors.success : AppColors.danger,
                    fontWeight: FontWeight.w600,
                  ),
                ),
              ),
            PrimaryButton(
              title: _loading ? 'Locating...' : 'Mark attendance',
              onPressed: _submit,
              loading: _loading,
            ),
          ],
        ),
      ],
    );
  }
}
